.github/workflows/check-cargo-check-runtimes.yml

```diff
@@ -0,0 +1,114 @@
+name: Check Cargo Check Runtimes
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened, ready_for_review, labeled ]
+
+
+# Jobs in this workflow depend on each other, only for limiting peak amount of spawned workers
+
+jobs:
+  # GitHub Actions allows using 'env' in a container context.
+  # However, env variables don't work for forks: https://github.com/orgs/community/discussions/44322
+  # This workaround sets the container image for each job using 'set-image' job output.
+  set-image:
+    if: contains(github.event.label.name, 'GHA-migration') || contains(github.event.pull_request.labels.*.name, 'GHA-migration')
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    outputs:
+      IMAGE: ${{ steps.set_image.outputs.IMAGE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - id: set_image
+        run: cat .github/env >> $GITHUB_OUTPUT
+  check-runtime-assets:
+    runs-on: arc-runners-polkadot-sdk-beefy
+    needs: [set-image]
+    container:
+      image: ${{ needs.set-image.outputs.IMAGE }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Run cargo check
+        uses: ./.github/actions/cargo-check-runtimes
+        with:
+          root: cumulus/parachains/runtimes/assets
+
+  check-runtime-collectives:
```

Comment:
How about to put the runtimes names in the matrix like:
matrix:
  runtimes: [assets, collectives, etc]
 to reduce amount of repetitive steps?

Reply:
We discussed that internally, there is [a comment](https://github.com/paritytech/polkadot-sdk/pull/5340/files#diff-ca6ddd9d3cdff8f696fe610915c9cd9316882794d8ed1e1e4dc6be392c0eb3e3R8). The main reason is that we have a certain amount of runners so using matrix uses them up faster. So instead running 9 jobs at once we run 2-3 jobs at one time and occupy less amount of runners.

Reply:
Oki, got it:)